scala-package/core/src/main/scala/ml/dmlc/mxnet/Optimizer.scala

```diff
@@ -139,4 +191,6 @@ trait MXKVStoreUpdater {
    */
   def update(key: Int, recv: NDArray, local: NDArray): Unit
   def dispose(): Unit
+  def serializeState(): Array[Byte]
+  def deserializeState(bytes: Array[Byte]): Unit
```

Comment:
I create another trait `MXKVStoreCachedStates`. I think it is more flexible since not all updaters need states.